src/controllers.php

```diff
@@ -163,6 +164,10 @@
         $link->setAttribute('class', 'anchor');
         $node->appendChild($link);
 
+        if (!isset($first_title) && strlen($title) > 0) {
```

Comment:
IMO, ``null === $firstTitle`` is better to decide whether we already have a title